Add Appveyor control files

pfmoore opened the pull request:
Note that these will not activate appveyor testing, as the pypa/pip project has not been registered with appveyor. However, it will allow forks (such as mine) to add appveyor testing without needing to maintain a separate branch to do so.

Ultimately it would be good to activate appveyor on the main repository, but I would rather wait until the tests pass, and we are sure that we can do so without interfering with the Travis builds.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/pypa/pip/2078)
<!-- Reviewable:end -->

techtonik commented:
+1

xavfernandez commented:
@pfmoore Why wasn't it merged ? This would be nice to have appveyor testing pip...

pfmoore commented:
@xavfernandez Honestly, I can't remember. But note that this *doesn't* activate Appveyor testing of pip itself, precisely because the tests don't run cleanly on Windows. My intention at one stage was to get the tests working on Windows (and that may have been what motivated this PR) but I ran out of spare time to work on that so it never got any further than an idea.

I don't mind merging it though, if anyone is interested. But it may be a little while before I can get to resolving the merge conflicts.

xavfernandez commented:
> But note that this doesn't activate Appveyor testing of pip itself, precisely because the tests don't run cleanly on Windows

Not sure I understand what you mean. The tests couldn't run ? Or some tests did fail ?

pfmoore commented:
Even with this PR, you still need to register the pip project with appveyor. Of course, we could easily do that, but if we did, then we'd hit the issue that (unless things have been fixed) a lot of the tests fail on Windows.

xavfernandez commented:
Failing tests are better than no idea what tests pass on windows from my point of view :)
Having appveyor would ease the acceptance of PR like: https://github.com/pypa/pip/pull/3133

pfmoore commented:
OK, I'll try to find some time to fix the conflicts, and merge it.

xavfernandez commented:
Great :) :+1:

pfmoore commented:
Actually, reviewing this, I'm not actually sure this PR is even needed any more - appveyor now includes Python 2.6-3.5 by default, so all the fun around installing Python is probably no longer needed. You could probably get away with a simple ```appveyor.yml``` nowadays. But it's difficult to confirm that as too many of pip's tests fail on Windows.

I'm inclined to drop this PR. The idea can be resurrected when someone has sufficient time to look at getting the tests in a workable state on Windows.

pfmoore commented:
Nowadays, Appveyor has all the CPython versions you'd want. So I've removed all the complexity and modified the PR to just include a simple ```appveyor.yml```.

I'll merge once Travis has completed.

Feel free to clone and enable Appveyor on your clone, and see if you can get the tests running cleanly. I would recommend *not* enabling Appveyor on the main pip repo until the tests run cleanly on Appveyor.

pfmoore commented:
OK, tests completed OK so I'll merge. If you want to see what a run under Appveyor looks like, check out https://ci.appveyor.com/project/pfmoore/pip/build/1.0.11 - it looks like there's a lot of Unix-specific assumptions in the test suite...

xavfernandez commented:
Thanks @pfmoore !
What would be the actions needed for appveyor to directly follow pip/develop ?

pfmoore commented:
If I recall, you need to sign up to appveyor under the pypa github account. Then add pip to the list of builds. That's basically it. What I don't know is how to use appveyor with a group account like pypa.

The downside is that every commit and PR will fail the appveyor tests, which may make it hard to see what PRs can be validly merged. I'm -1 on doing this until the tests worked.

For getting to the stage where the tests are OK, I'd suggest signing up on your personal github account, forking pip and adding your fork. You'll get the appveyor results, but they won't be visible in the main pip repo. Basically that's what I did at https://ci.appveyor.com/project/pfmoore/pip.

xavfernandez commented:
Hmm apparently you can tell AppVeyor to only build specific branches.
So maybe if you only tell him to build develop, it wont mark the PR as failing ?

I agree having failing AppVeyor markers on every PR is not desirable.

pfmoore commented:
One problem here is that when the 64-bit Python versions are used, we still set a TOXENV of (say) "py34", which picks up the 32-bit version. This isn't an issue for Travis, as it doesn't have both 32 and 64 bit versions. And in fact, it seems unlikely that we're ever likely to find an issue that *only* occurs on 64-bit Python and not on 32-bit, but for completeness I've just opened #3280 to use the "py" environment for tox, which says "use the version of Python you used to run tox". (The "py" environment was suggested by the author of tox - it's not documented (yet) but I'd say that makes it official enough for me :-))
